test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs

```diff
@@ -102,9 +102,187 @@ public void Configure_AllowPartialHostStartup()
             //Assert.False(scriptConfig.HostConfig.AllowPartialHostStartup);
         }
 
-        private ScriptHostOptionsSetup CreateSetupWithConfiguration(Dictionary<string, string> settings = null)
+        [Fact]
+        public void Configure_AppliesDefaults_IfDynamic()
+        {
+            var settings = new Dictionary<string, string>
+            {
+                { ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, LanguageWorkerConstants.LanguageWorkersSectionName, "maxMessageLength"), "2500" }
+            };
+
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSku, "Dynamic");
+
+            var options = GetConfiguredOptions(settings, environment);
+
+            Assert.Equal(ScriptHostOptionsSetup.DefaultFunctionTimeout, options.FunctionTimeout);
+            Assert.Equal(LanguageWorkerConstants.DefaultMaxMessageLengthBytesDynamicSku, options.MaxMessageLengthBytes);
+
+            // TODO: DI Need to ensure JobHostOptions is correctly configured
```

Comment:
Tracked by https://github.com/Azure/azure-functions-host/issues/3256

Reply:
A GA must have, right?

Reply:
Yes, I labeled it as a must.